pool/src/pool/actions.rs

```diff
@@ -207,6 +212,7 @@ pub fn build_actions_from_request(
             }
             RequestType::Repay => {
                 let mut reserve = pool.load_reserve(e, &request.address, true);
+                reserve.require_action_allowed(e, request.request_type);
```

Comment:
repaying assets is always allowed, regardless of any combination of pool or reserve status, thus this call is not needed.